### PR TITLE
chore: support parenthesized expressions in the java driver INTELLIJ-85

### PR DESCRIPTION
**Pull request by kmruiz**
### Checklist

- [ ] New tests and/or benchmarks are included.
- [ ] Documentation is changed or added.
- [ ] Changelog is updated accordingly.
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement).

### CODE REVIEW

File: packages/mongodb-dialects/java-driver/src/main/kotlin/com/mongodb/jbplugin/dialects/javadriver/glossary/PsiMdbTreeUtil.kt [hunk 1]
```diff
@@ -369,3 +375,17 @@ fun <T> PsiElement.findAllChildrenOfType(type: Class<T>): List<T> {
...
+/**
+ * Resolves to the first meaningful expression in a tree. Not all expressions have an important
+ * meaning for us (like for example, parenthesized expressions) so we drop them and get any
+ * inner expression that is relevant for us.
+ */
+fun PsiElement.meaningfulExpression(): PsiElement {
+    return when (this) {
+        // the children are: '(', YOUR_EXPR, ')'
+        // so we need the expression inside (index 1)
+        is PsiParenthesizedExpression -> children[1].meaningfulExpression()
```
Comment (himanshusinghs) [inline packages/mongodb-dialects/java-driver/src/main/kotlin/com/mongodb/jbplugin/dialects/javadriver/glossary/PsiMdbTreeUtil.kt:388]: Shouldn't we assume that this index access might result in a null value or an index out of bounds exception?

Comment (himanshusinghs) [reply packages/mongodb-dialects/java-driver/src/main/kotlin/com/mongodb/jbplugin/dialects/javadriver/glossary/PsiMdbTreeUtil.kt:388]: I believe this situation might come when the parenthesized expression is empty or incomplete?

Comment (kmruiz) [reply packages/mongodb-dialects/java-driver/src/main/kotlin/com/mongodb/jbplugin/dialects/javadriver/glossary/PsiMdbTreeUtil.kt:388]: Ah true, good point, we can either return children[1] or this. Making a quick fix.

**Review by himanshusinghs — APPROVED** [review]
🚀

**Pull request merged**